src/grafana_agent.py

```diff
@@ -219,6 +219,11 @@ def _on_cloud_config_revoked(self, _) -> None:
         self._update_config()
 
     # Abstract Methods
+    @property
+    def is_k8s(self) -> bool:
+        """Is this a k8s charm."""
+        raise NotImplementedError("Please override the is_k8s method")
```

Comment:
This can be implemented at the base, no need to inherit.

Reply:
How?